### PR TITLE
Update handling for Panel.collapsedTitle and add support for Panel.collapsedIcon

### LINKED ISSUE #3477: Update to `Panel.collapsedTitle` handling + support `collapsedIcon`
The new `Panel.collapsedTitle` prop would be useful in more cases if you could spec it on a panel that didn't have a "non-collapsed" title (or any other elements that trigger the display of panel header). That way you could easily have a panel show a header only when collapsed to aid in discovery/expand.

Along with that change, supporting `collapsedIcon` would provide the most common other option you might want to conditionally set on a panel header.

Finally, I'm thinking that `collapsedHeaderItems` would somehow a stretch too far - why would a user want to trigger header items on a panel where they can't see the content?  We currently always hide headerItems when collapsed for this reason, which seems sound to me.  If we introduced `collapsedHeaderItems` I think we'd need it to follow the conventions of the other props and default to `headerItems` if not specified, which means that apps would need to change their code to maintain the current behavior.  (So tldr let's leaver headerItems handling alone)

### PR DESCRIPTION
**Pull request by saba-mo**
Resolves #3477 

See Toolbox PR https://github.com/xh/toolbox/pull/671
 
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/panel/impl/PanelHeader.ts [hunk 1]
```diff
@@ -42,11 +44,11 @@ export const panelHeader = hoistCmp.factory({
             compactCls = compact ? 'xh-panel-header--compact' : null;
 
         // 1) Classic "top" title bar
-        if (!collapsed || vertical || isModal) {
+        if ((!collapsed || vertical || isModal) && (title || icon || headerItems)) {
```
Comment (amcclain) [inline desktop/cmp/panel/impl/PanelHeader.ts:47]: What is this change doing here?  I don't think we need it - think we want to be checking off of `displayedTitle/Icon`, and the change to the early-out on line 32 above should take care of the case where there aren't any of these.

Although note that we deliberately don't render `headerItems` when collapsed - see below line 60/62. I'm just reading here in Github, but it seems to me that we could setup `displayedHeaderItems` above to be an empty array if `collapsed && !isModal` - then we really have all of our collapsed-sensitive state consistently available and ready to check.

Also - sorry just looking a bit more closely - we should be careful that the early-out above isn't violating the expectations/requirements of the modal support stuff - it's planning to put a button in the header.  

I think this all needs to be looked at a bit more carefully

Comment (saba-mo) [reply desktop/cmp/panel/impl/PanelHeader.ts:47]: Now that I'm looking at it more closely, I'm not sure why we are showing the modal button in collapsed headers - in general but also specifically collapsed headers of top/bottom panels but not left/right panels. Why would someone want to pop out a collapsed panel? If it is something we want to continue to offer, shouldn't we do the same for all panels regardless of side?

Comment (TomTirapani) [reply desktop/cmp/panel/impl/PanelHeader.ts]: I think the aim there was to show a collapsed top/bottom header the same as an expanded one, with the same header tools and / or modal button. We can be reasonably sure that we have the same amount of space to work with and didn't want things to jump about. You can think of it as "the **content** goes away, and the existing header moves down".

Collapsing left / right is a different beast, as it changes the dimensions and orientation of the header. We can't reliably expect the available space to be equivalent - typically (as in the Portfolio example app) we'll have a lot less space to work with. You can think of it as "the **panel** goes away, and is replaced by a minimal affordance". Rotated text is harder to read, and rotated header items / buttons are just not particularly usable. There are also technical styling reasons for not attempting to rotate header items.

At least that's how I think about it 🤷‍♂️

**Comment by TomTirapani** [comment]
Regarding the modal button - it seems the question we need to ask is "should we show a header with *only* a modal button?".

Right now (both develop and this PR) the answer is no - you need at least one other item in your header for it to show at all. However, given that we are willing to show a header with **only** header items (and you can think of the modal button as built in header item), this is likely an oversight on our part.

I think we should aim for needing at least one of Title / Icon / Header Items / Modal Button required to display. What do you think?

**Comment by saba-mo** [comment]
I agree if header items are enough, the modal button should be enough to show a header. But if you pop out a collapsed panel with only a modal button, you end up with this (bottom panel is modal), which definitely looks strange. However, in my opinions, it's not much stranger than the current ability to pop out a collapsed panel. I'm not sure how complicated this would be, but what do you think about toggling a panel's collapsed status when clicking the modal button so it's in an expanded state and shows its content? 
Screenshot 2023-09-14 at 10 45 05 AM
Screenshot 2023-09-14 at 10 45 11 AM

File: desktop/cmp/panel/impl/PanelHeader.ts (hunk 1 above)
Comment (lbwexler) [reply desktop/cmp/panel/impl/PanelHeader.ts]: I like your way of thinking about it tom

Comment (amcclain) [reply desktop/cmp/panel/impl/PanelHeader.ts]: In new PR #3480 I went with consistently showing the modalToggle for both LR and TB collapsed panels - that seemed like a pretty clear choice to me, as long as we're OK with both buttons migrating to the bottom of the vertical header.

I still left out app-specific headerTools for collapsed panels, even TB-collapsed ones with the horizontal header. My thinking was that it's very likely that headerTools are only there to control something about the content of the panel, and it seemed potentially dangerous (or at least weird) to show them when you can't see that content.  

If we hit a specific use case where we do want them, we could consider adding a flag to control at that point - `showHeaderToolsWhenCollapsed` or something crazy like that :)

**Comment by amcclain** [comment]
Closing in favor of #3480 which includes additional fixes and refactoring

**Pull request closed without merging**